recipe/build.sh

```diff
@@ -2,21 +2,22 @@
 
 set -ex
 
-if [[ "$CONDA_BUILD_CROSS_COMPILATION" == 1 ]]; then
+if [[ "${target_platform}" == osx* ]]; then
+  export CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_CXX_STANDARD=14"
+else
+  export CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_CXX_STANDARD=17"
+fi
```

Comment:
Is there a technical reason for this distinction? Will it not build on Linux with gcc under C++14? I see that we are explicitly selecting a more recent compiler below. Wouldn't it be better to keep a lowest common denominator across platforms?

Reply:
I've added documentation in the file to explain the need - it has to do with how `abseil-cpp` is built.